personalized_nlp/learning/train.py

```diff
@@ -23,6 +24,9 @@ def train_test(datamodule, model, epochs=6, lr=1e-2, regression=False,
         if isinstance(datamodule.annotation_column, str):
             class_names = [datamodule.annotation_column]
 
+        if not model._frozen and weight_decay!=0.0:
+          model = RegressorFinetune(model=model, lr=lr, class_names=class_names)
+
```

Comment:
It is related with the seperate class `RegressorFinetune`, adding new class is less flexible. For example, here `model` object does not have to have `_frozen` field and in that case there will be an error.

Another suggestion abot code style: `_` prefix suggests it is a private filed so it should not be accessed outside the class.

Reply:
Yes, at first I thought that `_frozen` could be defined as private use, but I fix it to make it more flexible.
I also resolve the problem with `RegressorFinetune`, by using callbacks instead. Now there's no overwritting.